x-pack/plugins/alerts/server/alerts_client/alerts_client.ts

```diff
@@ -1374,6 +1382,15 @@ export class AlertsClient {
     }
   }
 
+  private validateAlertNotifyWhenType(
```

Comment:
nit.
AlertClient is getting way too big.
Any chance we can extract this somewhere else?
Moving it will also make it less painful to add some tests, as nested turneries are error prone and easy to break accidentally (I'm guilty of this as well :) )

Reply:
Extracted into its own file and added tests.